lib/src/facts/windows/operating_system_resolver.cc

```diff
@@ -101,7 +104,25 @@ namespace facter { namespace facts { namespace windows {
         auto version = result.release.substr(0, lastDot);
         bool consumerrel = (wmi::get(vals, wmi::producttype) == "1");
         if (version == "10.0") {
-            result.release = consumerrel ? "10" : "2016";
+            // Calculate the build number to distinguish between
+            // Windows Server 2016 and 2019. Note that the kernel
+            // version is written as <major>.<minor>.<build_number>
+            auto kernel_version_fact = facts.get<string_value>(fact::kernel_version);
```

Comment:
I figured calculating the build number from the kernel version fact is easier because calling RtlGetVersion requires [several steps](https://github.com/puppetlabs/facter/blob/master/lib/src/facts/windows/kernel_resolver.cc#L19-L34) and I didn't see a need to create a common function when kernel version gets this info. for us (and the string parsing isn't complicated).

Reply:
I can't think of a reason why this is not-good.  Is there a precedence for dependent core facts?

Reply:
No, we use dependent core facts elsewhere too like e.g. [here](https://github.com/puppetlabs/facter/blob/master/lib/src/facts/aix/operating_system_resolver.cc#L89). My only concern with doing it this way was b/c it isn't a direct C API call to `RtlGetVersion` (which is the general pattern for obtaining these facts in cfacter).